From cd769dab4ee32a7a74e6aa2716b6ea55c05a0b60 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Mon, 28 Apr 2025 01:43:42 +0200
Subject: [PATCH 077/115] dt-bindings: thermal: mediatek: lvts-thermal: Add
 support for MT6785

Add mediatek,mt6785-lvts compatible and thermal bindings.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../bindings/thermal/mediatek,lvts-thermal.yaml    | 14 ++++++++++++++
 .../dt-bindings/thermal/mediatek,lvts-thermal.h    |  9 +++++++++
 2 files changed, 23 insertions(+)

diff --git a/Documentation/devicetree/bindings/thermal/mediatek,lvts-thermal.yaml b/Documentation/devicetree/bindings/thermal/mediatek,lvts-thermal.yaml
index 975235130..09072559b 100644
--- a/Documentation/devicetree/bindings/thermal/mediatek,lvts-thermal.yaml
+++ b/Documentation/devicetree/bindings/thermal/mediatek,lvts-thermal.yaml
@@ -18,6 +18,7 @@ description: |
 properties:
   compatible:
     enum:
+      - mediatek,mt6785-lvts
       - mediatek,mt7987-lvts-ap
       - mediatek,mt7988-lvts-ap
       - mediatek,mt8186-lvts
@@ -94,6 +95,19 @@ allOf:
         nvmem-cell-names:
           minItems: 2
 
+  - if:
+      properties:
+        compatible:
+          contains:
+            const: mediatek,mt6785-lvts
+    then:
+      properties:
+        nvmem-cells:
+          minItems: 3
+
+        nvmem-cell-names:
+          minItems: 3
+
 required:
   - compatible
   - reg
diff --git a/include/dt-bindings/thermal/mediatek,lvts-thermal.h b/include/dt-bindings/thermal/mediatek,lvts-thermal.h
index 350f98178..1f8c6e8a3 100644
--- a/include/dt-bindings/thermal/mediatek,lvts-thermal.h
+++ b/include/dt-bindings/thermal/mediatek,lvts-thermal.h
@@ -7,6 +7,15 @@
 #ifndef __MEDIATEK_LVTS_DT_H
 #define __MEDIATEK_LVTS_DT_H
 
+#define MT6785_BIG_CPU0	0
+#define MT6785_BIG_CPU1	1
+#define MT6785_LITTLE_CPU0	2
+#define MT6785_LITTLE_CPU1	3
+#define MT6785_LITTLE_CPU2	4
+#define MT6785_GPU0		5
+#define MT6785_GPU1		6
+#define MT6785_VPU		7
+
 #define MT7987_CPU		0
 #define MT7987_ETH2P5G		1
 
-- 
2.56.0

